### PR TITLE
Draft for advancement opportunities in 2025

### PR DESCRIPTION
**Pull request by dotsdl**
This is our first advancement opportunities perspective. It functions as a vision for the future, and a starting point for coordinating effort and resources to pursue it.

### CODE REVIEW

**Comment by dotsdl** [comment]
@ianmkenney can I get a review from you on this? It's ready to go!

File: content/opportunities/alchemiscale-2025/index.md [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+---
+
+This is a perspective on where we can take the **alchemiscale** project in 2025, given how far we've come and what is currently in development.
+If you have any questions on the content, would like to get involved in development, or would like to sponsor any of these advancements, get in touch on our [Discussions thread]()!
+
+
+## where we've been
+
+We've come a long way since the **alchemiscale** project kicked off in early 2022.
+Originally concieved as a wrapper around [**Folding@Home**](https://foldingathome.org/) for performing alchemical binding free energy calculations routinely and at scale, the project co-evolved with the [**Open Free Energy (OpenFE)**](https://openfree.energy/) data model to become a general-purpose, high-throughput execution system that can take advantage of a variety of distributed compute.
```
Comment (ianmkenney) [inline content/opportunities/alchemiscale-2025/index.md]:
```suggestion
Originally conceived as a wrapper around [**Folding@Home**](https://foldingathome.org/) for performing alchemical binding free energy calculations routinely and at scale, the project co-evolved with the [**Open Free Energy (OpenFE)**](https://openfree.energy/) data model to become a general-purpose, high-throughput execution system that can take advantage of a variety of distributed compute.
```

File: content/opportunities/alchemiscale-2025/index.md [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+
+#### additional `Strategy` implementations
+
+Beyond [NetBFE](https://pubs.acs.org/doi/10.1021/acs.jctc.1c00703), there are other `Strategy`s one could implement that optimize for different things.
+These include choosing to allocate compute based on:
+
+1. `Transformation`/`Mapping` score
+2. critical connectivity in the `AlchemicalNetwork`
+
+Exploring thes space of possibilities for different `Strategy`s would add richness to the portfolio of adaptive sampling approaches available.
```
Comment (ianmkenney) [inline content/opportunities/alchemiscale-2025/index.md]: this space

File: content/opportunities/alchemiscale-2025/index.md [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+Users can also experiment with `Strategy`s of their own locally, and enabling this via the `AlchemiscaleClient` is something we would like to spend effort on.
+
+
+#### `Strategy` stacking
+
+We are currently working on iniitial support for `Strategy`s (see [**where we are**]({{< ref "#where-we-are" >}})), but beyond this there may be a good case for being able to *stack* multiple `Strategy`s on a single `AlchemicalNetwork`.
```
Comment (ianmkenney) [inline content/opportunities/alchemiscale-2025/index.md]: initial

File: content/opportunities/alchemiscale-2025/index.md [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+as these are not preserved by **alchemiscale**.
+A challenge with supporting result file retention and retrieval is the potentially unbounded size of result files produced by a given `Protocol`,
+with some `Protocol`s routinely producing files 10s of GiB in size with default settings choices.
+If **alchemiscale** is to support retaining these files and making them available to users via the `AlchemiscaleClient`, the approach taken must put sane constraints on what is preserved and exposed.
+
+One approach that could work well is implemented in [gufe#186](https://github.com/OpenFreeEnergy/gufe/pull/186).
+This involves adding a third class of storage to the `Protocol` system, `permanent`, that `Protocol` authors can use for file outputs they would like preserved by the execution system.
+This allows `scratch` and `shared` storage to continue to persist over the life of individual `ProtocolUnit`s and the `ProtocolDAG`, respectively, which may require much larger files to be produced during simulation runs.
+It also gives a very clear signal to the execution system what should be preserved post-`ProtocolDAG` execution, requring no heuristics or guessing.
```
Comment (ianmkenney) [inline content/opportunities/alchemiscale-2025/index.md]: requiring

File: content/opportunities/alchemiscale-2025/index.md [hunk 5]
```diff
@@ -0,0 +1,403 @@
...
+
+#### prediction of `AlchemicalNetwork` computational cost accounting for existing `Transformation` results
+
+In addition to the above, since `AlchemicalNetwork`s submitted to the same `Scope` share `Transformation`s, and therefore results, the estimator could be made to only consider `Transformation`s that don't exist in the `Scope` a user intends to submit their `AlchemicalNetwork` into.
+This could help to give more accurate predictions of cost, especially for highly-iterative usage patterns like [**living networks**]({{< ref "#living-networks" >}}).
+
+
+### operations
+
+In addition to the other areas above, there are more "boring" areas that could greatly improve **alchemiscale**'s operations model.
```
Comment (ianmkenney) [inline content/opportunities/alchemiscale-2025/index.md]: In addition to the other areas mentioned above, there are several key areas that could significantly enhance alchemiscale's operations model.

File: content/opportunities/alchemiscale-2025/index.md [hunk 6]
```diff
@@ -0,0 +1,403 @@
...
+
+#### tooling for performing data model migrations
+
+**alchemiscale** is an *execution system*, not an *archival system*, and this distinction loosely encodes our guarantees to users on result retrievability as the data models **alchemiscale** depends on evolve.
+We currently have very little if any migration machinery in place for transitioning data models in an existing **alchemiscale** instance to new versions featured in e.g. [**openfe**](https://github.com/OpenFreeEnergy/openfe) or [**feflow**](https://github.com/OpenFreeEnergy/feflow).
+
+We should formalize the boundaries of our guarantees around support for data model updates in **alchemiscale** itself and common packages deployed with it,
+then build a framework for updating data models on deployed systems in compliance with those guarantees.
```
Comment (ianmkenney) [inline content/opportunities/alchemiscale-2025/index.md]: inconsistent line break, if you care

File: content/opportunities/alchemiscale-2025/index.md [hunk 7]
```diff
@@ -0,0 +1,403 @@
...
+If any of these areas sound appealing to you, your help is more than welcome!
+Get in touch on our [Discussions thread](), and we'd be happy to help get you started.
+
+
+### visualization
+
+**alchemiscale** features a powerful Python client that enables users to submit `AlchemicalNetwork`s, action and prioritize `Transformation`s for compute, and pull results as they become available.
+However, rich visualization of the state of **alchemiscale** and its calculations is not something the client will ever be capable of, given its nature as a simple HTTP interface.
+
+Administrators are even worse-off currently, as managing users on an **alchemiscale** instance requires execution of CLI commands against the server, which is often awkward and tedious. Tracking compute throughput also requires direct Cypher queries on the Neo4j database backend, requiring knowledge of Cypher to do this effectively, while also presenting the risk of destructive actions that could put the system in an unrecoverable, inconsistent state.
```
Comment (ianmkenney) [inline content/opportunities/alchemiscale-2025/index.md]: link to cypher

**Review by ianmkenney — CHANGES_REQUESTED** [review]
Looks good overall, but there are a few small changes!

**Comment by dotsdl** [comment]
Thanks @ianmkenney! I've implemented all your requested changes!

**Pull request merged**